pkg/controller.v2/trainjob_controller.go

```diff
@@ -199,7 +200,7 @@ func setSuspendedCondition(trainJob *kubeflowv2.TrainJob) {
 
 func setTerminalCondition(ctx context.Context, runtime jobruntimes.Runtime, trainJob *kubeflowv2.TrainJob) error {
 	terminalCond, err := runtime.TerminalCondition(ctx, trainJob)
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Instead of this, I'm wondering if we should update the status regardless of errors there: https://github.com/kubeflow/training-operator/blob/3eb1a0c9e3616a7fed9864d8b1373d6d677a6c77/pkg/controller.v2/trainjob_controller.go#L99-

So, if we get any error in L100https://github.com/kubeflow/training-operator/blob/3eb1a0c9e3616a7fed9864d8b1373d6d677a6c77/pkg/controller.v2/trainjob_controller.go#L96-L98. we just join the `terminalCondErr` with `error`, but we do not return error here.

In other words, we update the status in the case of status differences between old and new even if we get any errors from setCondition functions.

Because we do not restrict underlying Job to JobSet and we can not estimate that the `NotFound` error can be ignored.

Reply:
You're right, that's much better!

I've just re-pushed. Let me know if that corresponds to what you had in mind.